packages/eslint-plugin/src/rules/thenable-in-promise-aggregators.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  isTypeAnyType,
+  isTypeUnknownType,
+} from '@typescript-eslint/type-utils';
+import type { TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+
+import { createRule, getParserServices } from '../util';
+
+export default createRule({
+  name: 'thenable-in-promise-aggregators',
+  meta: {
+    docs: {
+      description:
+        'Disallow passing non-Thenable values to promise aggregators',
+      recommended: 'recommended',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      inArray:
+        'Unexpected non-Thenable value in array passed to promise aggregator.',
+      arrayArg:
+        'Unexpected array of non-Thenable values passed to promise aggregator.',
+      nonArrayArg: 'Unexpected non-array passed to promise aggregator.',
+    },
+    schema: [],
+    type: 'problem',
+  },
+  defaultOptions: [],
+
+  create(context) {
+    const services = getParserServices(context);
+    const checker = services.program.getTypeChecker();
+
+    return {
+      CallExpression(node: TSESTree.CallExpression): void {
+        if (node.callee.type !== AST_NODE_TYPES.MemberExpression) {
+          return;
+        }
+        if (node.callee.object.type !== AST_NODE_TYPES.Identifier) {
+          return;
+        }
+        if (node.callee.object.name !== 'Promise') {
+          return;
+        }
+        if (node.callee.property.type !== AST_NODE_TYPES.Identifier) {
+          return;
+        }
+
+        const { name } = node.callee.property;
+        if (!['race', 'all', 'allSettled'].includes(name)) {
+          return;
+        }
+
+        const { arguments: args } = node;
+        if (args.length !== 1) {
+          return;
+        }
+
+        const arg = args[0];
+        if (arg.type === AST_NODE_TYPES.ArrayExpression) {
+          const { elements } = arg;
+          if (elements.length === 0) {
+            return;
+          }
+
+          for (const element of elements) {
+            if (element == null) {
+              continue;
+            }
```

Comment:
We don't have test case containing `Promise.all([,])`, let's add it?

Also not sure if this code should be reported or not? Technically `null` is not a thenable

Reply:
Ahh I wasn't really clear what this one was testing for I just saw it in several other tests. Thank you! Reporting & adding as a test